dspace-api/src/main/java/org/dspace/util/SolrImportExport.java

```diff
@@ -0,0 +1,583 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.util;
+
+import org.apache.commons.cli.*;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.log4j.Logger;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrServer;
+import org.apache.solr.client.solrj.request.AbstractUpdateRequest;
+import org.apache.solr.client.solrj.request.ContentStreamUpdateRequest;
+import org.apache.solr.client.solrj.request.CoreAdminRequest;
+import org.apache.solr.client.solrj.request.LukeRequest;
+import org.apache.solr.client.solrj.response.LukeResponse;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.luke.FieldFlag;
+import org.apache.solr.common.params.CoreAdminParams;
+import org.dspace.core.ConfigurationManager;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * Utility class to export, clear and import Solr indexes.
+ * @author Andrea Schweer schweer@waikato.ac.nz for the LCoNZ Institutional Research Repositories
+ */
+public class SolrImportExport
+{
+
+	private static final DateFormat SOLR_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+	private static final DateFormat EXPORT_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd-HH-mm");
+
+	static
+	{
+		SOLR_DATE_FORMAT.setTimeZone(TimeZone.getTimeZone("UTC"));
+		EXPORT_DATE_FORMAT.setTimeZone(TimeZone.getDefault());
+	}
+
+	private static final String ACTION_OPTION = "a";
+	private static final String CLEAR_OPTION = "c";
+	private static final String DIRECTORY_OPTION = "d";
+	private static final String HELP_OPTION = "h";
+	private static final String INDEX_NAME_OPTION = "i";
+	private static final String KEEP_OPTION = "k";
+	private static final String LAST_OPTION = "l";
+
+	public static final int ROWS_PER_FILE = 10_000;
```

Comment:
I realize this is set to 10K to make the process go faster, but it's fairly demanding of a server that may be doing other things at the same time. I am going to try setting this number lower and see how that works out.